checks/system/slurm/gres.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2016-2023 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.simple_test
+class SlurmGresTest(rfm.RunOnlyRegressionTest):
+    valid_systems = ['+scontrol']
+    valid_prog_environs = ['builtin']
+    num_tasks_per_node = 1
+    partition = variable(str, value='nvgpu')
+    gres = variable(str, value='gpu:4')
```

Comment:
Does it make sense to get the number of gpus from the partition devices?

Reply:
Also, I am noticing that in the amd partition we don't have info about the devices. Maybe we could add it in this PR?

Reply:
Hmm. That's a good idea. I was just thinking whether to leave it general like that (so that the gres variable can be passed by the user) or make it more specific like `GPUGresTest` , what do you think?

Reply:
Hm okay, both are useful I think. How about keeping `SlurmGresTest` as a base class and then specializing to `GPUGresTest` where the value of the variable will be overwritten to the number of gpus that are available through the config?
They way we structure the repo right now, you would have to remove the `@rfm.simple_test` from the base class so that it doesn't run twice but maybe we could introduce some kind of "generic" library tests that are run directly with `-S` options in the future. Or just add tests like this in the new repo that we will make.